File: win-prog/ProgWinCS/Buttons and Labels and Scrolls (Oh, My!)/TwoButtons/TwoButtons.cs
//-----------------------------------------
// TwoButtons.cs © 2001 by Charles Petzold
//-----------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class TwoButtons: Form
{
     readonly Button btnLarger, btnSmaller;
     readonly int    cxBtn, cyBtn, dxBtn;

     public static void Main()
     {
          Application.Run(new TwoButtons());
     }
     public TwoButtons()
     {
          Text = "Two Buttons";
          ResizeRedraw = true;

          cxBtn = 5 * Font.Height;
          cyBtn = 2 * Font.Height;
          dxBtn =     Font.Height;

          btnLarger = new Button();
          btnLarger.Parent = this;
          btnLarger.Text   = "&Larger";
          btnLarger.Size   = new Size(cxBtn, cyBtn);
          btnLarger.Click += new EventHandler(ButtonOnClick);

          btnSmaller = new Button();
          btnSmaller.Parent = this;
          btnSmaller.Text   = "&Smaller";
          btnSmaller.Size   = new Size(cxBtn, cyBtn);
          btnSmaller.Click += new EventHandler(ButtonOnClick);

		 OnResize(EventArgs.Empty);
     }
     protected override void OnResize(EventArgs ea)
     {
          base.OnResize(ea);

          btnLarger.Location =
                         new Point(ClientSize.Width / 2 - cxBtn - dxBtn / 2,
                                  (ClientSize.Height - cyBtn) / 2);
          btnSmaller.Location =
                         new Point(ClientSize.Width / 2 + dxBtn / 2,
                                  (ClientSize.Height - cyBtn) / 2);
     }
     void ButtonOnClick(object obj, EventArgs ea)
     {
          Button btn = (Button) obj;

          if (btn == btnLarger)
          {
               Left   -= (int)(0.05 * Width);
               Top    -= (int)(0.05 * Height);
               Width  += (int)(0.10 * Width);
               Height += (int)(0.10 * Height);
          }
          else
          {
               Left   += (int)(Width  / 22f);
               Top    += (int)(Height / 22f);
               Width  -= (int)(Width  / 11f);
               Height -= (int)(Height / 11f);
          }
     }
}

File: win-prog/ProgWinVB/Buttons and Labels and Scrolls (Oh, My!)/TwoButtons/TwoButtons.vb
'-------------------------------------------
' TwoButtons.vb (c) 2002 by Charles Petzold
'-------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class TwoButtons
    Inherits Form

    ReadOnly btnLarger, btnSmaller As Button
    ReadOnly cxBtn, cyBtn, dxBtn As Integer

    Shared Sub Main()
        Application.Run(New TwoButtons())
    End Sub

    Sub New()
        Text = "Two Buttons"
        ResizeRedraw = True

        cxBtn = 5 * Font.Height
        cyBtn = 2 * Font.Height
        dxBtn = Font.Height

        btnLarger = New Button()
        btnLarger.Parent = Me
        btnLarger.Text = "&Larger"
        btnLarger.Size = New Size(cxBtn, cyBtn)
        AddHandler btnLarger.Click, AddressOf ButtonOnClick

        btnSmaller = New Button()
        btnSmaller.Parent = Me
        btnSmaller.Text = "&Smaller"
        btnSmaller.Size = New Size(cxBtn, cyBtn)
        AddHandler btnSmaller.Click, AddressOf ButtonOnClick

        OnResize(EventArgs.Empty)
    End Sub

    Protected Overrides Sub OnResize(ByVal ea As EventArgs)
        MyBase.OnResize(ea)
        btnLarger.Location = _
                New Point(ClientSize.Width \ 2 - cxBtn - dxBtn \ 2, _
                         (ClientSize.Height - cyBtn) \ 2)
        btnSmaller.Location = _
                New Point(ClientSize.Width \ 2 + dxBtn \ 2, _
                         (ClientSize.Height - cyBtn) \ 2)
    End Sub

    Private Sub ButtonOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Dim btn As Button = DirectCast(obj, Button)

        If btn Is btnLarger Then
            Left -= CInt(0.05 * Width)
            Top -= CInt(0.05 * Height)
            Width += CInt(0.1 * Width)
            Height += CInt(0.1 * Height)
        Else
            Left += CInt(Width / 22)
            Top += CInt(Height / 22)
            Width -= CInt(Width / 11)
            Height -= CInt(Height / 11)
        End If
    End Sub
End Class
